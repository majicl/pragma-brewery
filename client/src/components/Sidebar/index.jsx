import React from 'react';

export default ({ menuName, menuItems = [] }) => {
  return (
    <nav className="sidebar sidebar-offcanvas" id="sidebar">
      <ul className="nav">
        <li className="nav-item nav-category">{menuName}</li>

        {menuItems.map((item, index) => (
          <li className="nav-item" key={index}>
            <a className="nav-link" href={item.uri}>
              <span className="icon-bg">
                <i className="mdi mdi-cube menu-icon" />
              </span>
              <span className="menu-title">{item.title}</span>
            </a>
          </li>
        ))}
      </ul>
    </nav>
  );
};
