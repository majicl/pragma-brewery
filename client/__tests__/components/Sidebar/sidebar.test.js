import React from 'react';
import '@testing-library/jest-dom';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { render, queryHelpers } from '@testing-library/react';
import Sidebar from '~/Sidebar/index.jsx';

configure({ adapter: new Adapter() });
const queryByClassName = queryHelpers.queryByAttribute.bind(null, 'class');

describe('<Sidebar />', () => {
  it('matches snapshot on responsive', () => {
    const { container } = render(<Sidebar />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('The category navigation is empty if MenuName is not set', () => {
    const {
      container: { firstChild: sidebar }
    } = render(<Sidebar />);

    const menuTag = queryByClassName(sidebar, 'nav-item nav-category');
    expect(menuTag).not.toBeNull();
    expect(menuTag).toBeEmptyDOMElement();
  });

  it('The category navigation matches the MenuName', () => {
    const menuName = 'Dashboard';
    const {
      container: { firstChild: sidebar }
    } = render(<Sidebar menuName={menuName} />);

    const menuTag = queryByClassName(sidebar, 'nav-item nav-category');
    expect(menuTag).not.toBeNull();
    expect(menuTag).toHaveTextContent(menuName);
  });

  it('The empty MenuItems does not render any menu', () => {
    const {
      container: { firstChild: sidebar }
    } = render(<Sidebar />);

    const navTags = queryByClassName(sidebar, 'nav-item');
    expect(navTags).toBeNull();
  });

  it('The items in MenuItems render in menus', () => {
    const menuItems = [
      {
        title: 'Dashboard',
        uri: '/'
      },
      {
        title: 'Beers',
        uri: '/beers'
      }
    ];
    const sidebar = mount(<Sidebar menuItems={menuItems} />);
    expect(sidebar.find('.nav-item')).toHaveLength(3);
  });
});
