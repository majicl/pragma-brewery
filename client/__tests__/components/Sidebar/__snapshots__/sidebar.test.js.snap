// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> matches snapshot on responsive 1`] = `
<nav
  class="sidebar sidebar-offcanvas"
  id="sidebar"
>
  <ul
    class="nav"
  >
    <li
      class="nav-item nav-category"
    />
  </ul>
</nav>
`;
